Dictionary Service Completed and Translation started

kunal-mohta opened the pull request:
For **Google Summer of Code - 2018** - Evaluation Phase 2 & 3

kunal-mohta committed:
using kettle resolvers for external data

kunal-mohta committed:
oxford authorization error message bug fix

kunal-mohta committed:
pronunciations endpoint

kunal-mohta committed:
linting

kunal-mohta committed:
implementing %directory like paths

kunal-mohta committed:
basic frequency endpoint implementation; linting; readme update

kunal-mohta committed:
tests for frequency endpoint

kunal-mohta committed:
adding alltests.js and refactoring tests

kunal-mohta committed:
extended frequency endpoint implementation

kunal-mohta committed:
Tests for extended frequency service and update readme

kunal-mohta committed:
adding mock servers of oxford service for test mocking using nock

kunal-mohta committed:
basic compromise implementation; readme update; linting

kunal-mohta committed:
refactoring complex iterations with infusion core api

kunal-mohta committed:
breaking handlers file into shorter parts; ability to run all services on same port

kunal-mohta committed:
refactor wiktionary grades to have a common grade for services not provided

kunal-mohta committed:
compromise error handling; extracting common handler functions in handlerutils

kunal-mohta committed:
correcting variable name denoting the service name

kunal-mohta committed:
unit tests for wiktionary service and and some refactoring along

kunal-mohta committed:
compromise refactoring and unit tests; few oxford unit tests

kunal-mohta committed:
fixing alltests.js error due to nock

kunal-mohta committed:
refactoring existing unit tests

kunal-mohta committed:
completing unit tests

kunal-mohta committed:
imporiving directory structure of dictionary tests

kunal-mohta committed:
compromise contract test; some refactoring and directory structure change

kunal-mohta committed:
compromise contract test sample structure; useful utility functions for testing

kunal-mohta committed:
refactor construct response with better js methods

kunal-mohta committed:
contract tests - wiktionary and compromise

kunal-mohta committed:
contract tests complete - oxford and wiktionary

kunal-mohta committed:
unit tests - oxford constructResponse functions

kunal-mohta committed:
refactor oxford unit tests and nock files with common mock data files

kunal-mohta committed:
variable declaration refactor

kunal-mohta committed:
starting off translation service

simple yandex translation endpoint started; checkSourceText function with its unit test; nock for the endpoint

kunal-mohta committed:
commands for running tests; README update

kunal-mohta committed:
yandex translation endpoint

checkServiceKey function and unit test; requiredData function; nock servers for the endpoint

kunal-mohta committed:
yandex translation endpoint

all functions complete; tests left

kunal-mohta committed:
yandex translation service - unit tests complete

kunal-mohta committed:
yandex tests complete

contract and integration tests

kunal-mohta committed:
refactor translation tests to have a common mock data source

kunal-mohta committed:
readme update

kunal-mohta committed:
refactor oxford handlers

common requiredData function for all endpoints

kunal-mohta committed:
yandex language detection endpoint start

kunal-mohta committed:
tests for yandex language detection endpoint

kunal-mohta committed:
using kettle resolvers to access env variable, where not already used

kunal-mohta committed:
tests for long sentences/text in compromise and yandex

kunal-mohta committed:
handling empty lang responses for yandex language detection

handlers + tests update

kunal-mohta committed:
README update

waharnum commented on README.md:
````diff
@@ -2,27 +2,48 @@
 ```
 npm install
 ```
-**Step 3 - Setting API keys for services that require it**\
````
I believe this should be updated with instructions for adding the Yandex key.

kunal-mohta replied:
Sorry, my bad.
Will add instructions for Yandex.

waharnum replied:
One thing I'd suggest (I commonly do this for projects that need a secrets-type config file) is to create a sample `.env` file called something like `.env.example` that shows the expected structure and key names, without actually containing real secrets. Something like:

```
OXFORD_APP_ID=[Oxford app id here]
OXFORD_APP_KEY=[Oxford app key here]
YANDEX_APP_KEY=[Yandex app key here]
```

You can then refer to this in the documentation and commit it to version control - this can help in making sure your config file documentation is up to date.

kunal-mohta replied:
Great idea. Will surely do the same here.

waharnum commented on README.md:
```diff
@@ -51,18 +83,99 @@ Currently available endpoints -
+- `http://localhost:8083/v1/translation/yandex/detect`\
+**Method**: `POST`\
+**Request Body Format**: `{ text: "text for which language is to be detected"}`\
+**Language Code**: ISO 639-1 (2-lettered language codes)
```
Is the language code still relevant here, given this endpoint is meant to detect the text language? Also, is there a way to configure the language you want it translated into?

kunal-mohta replied:
That is supposed to be the the Language code standards for the one that is returned.
For this endpoint, it is not translating the text. It is just detecting its language.
The next endpoint, after this, is supposed to just take the target language for translation, detecting the language on its own.
More on this in our call.

waharnum replied:
Ah, got it - my misunderstanding. So this would probably be a two-part flow where text in an unknown language is detected, then that language code is used as part of a follow-up translation request.

waharnum commented on README.md:
```diff
@@ -51,18 +83,99 @@ Currently available endpoints -
+There are 3 kinds of tests for each service -
+- **Unit Tests** - Tests for individual functions
+- **Integration Tests** - Testing the overall functioning of the endpoint (using mock test servers)
+- **Contract Tests** - Testing if the external service provides data with the expected structure
```
Regarding the contract tests (and generally) - it would be useful to have some form of error detection and flagging if a needed key does not exist, perhaps when starting a service known to depend on that key.

waharnum reviewed (commented):
A few comments on the current README

kunal-mohta committed:
README update

instructions to acquire and use api keys

kunal-mohta committed:
error/warning for when the api keys are absent

terminate contract tests and display a warning when starting the service

kunal-mohta committed:
yandex translation with only target language given

kunal-mohta committed:
refactor error logs

kunal-mohta committed:
yandex handlers refactoring

kunal-mohta committed:
refactor oxford handlers

reduce repitition and add preRequest error check

kunal-mohta committed:
oxford tests update

updated and added tests based on refactoring

kunal-mohta committed:
yandex translation (with target lang only) tests

kunal-mohta committed:
oxford languages endpoint

kunal-mohta committed:
google translation start

detect and translate endpoint complete

kunal-mohta committed:
google lang detection endpoint - handlers

kunal-mohta committed:
google translation unit tests & translation test directory struct change

kunal-mohta committed:
contract tests for google translation

kunal-mohta committed:
google translation - get all supported languages endpoint

kunal-mohta committed:
unit and contract test for google translation get supported languages

kunal-mohta committed:
translation url fix; README update

kunal-mohta committed:
yandex list all supported languages endpoint, with unit and contract tests

kunal-mohta committed:
oxford list supported languages endpoint with unit and integration tests

kunal-mohta committed:
refactor oxford mock data and nock servers

adding instances for authentication errors to both

kunal-mohta committed:
update nock servers for translation; linting

kunal-mohta committed:
refactor oxford mock data usage

kunal-mohta committed:
yandex list languages integration test

kunal-mohta committed:
refactor dictionary schemas (for contract tests)

kunal-mohta committed:
update oxford integration tests and nock servers according to changes in mock data

kunal-mohta committed:
external service specifictions and README update

kunal-mohta committed:
integration tests for all google translation endpoints

kunal-mohta committed:
table of contents in README

kunal-mohta committed:
removing unnecessary dependencies and registered namespaces

kunal-mohta committed:
Wiktionary list supported languages endpoint

kunal-mohta committed:
using mock data for wiktionary integraion tests

for definition endpoint only (other endpoints not supported in wiktionary)

kunal-mohta committed:
Google translation extended list languages endpoint

handlers and extending the existing list languages tests for extended endpoint

kunal-mohta committed:
change response of list languages endpoints for all the services to maintain consistency

kunal-mohta committed:
update yandex list languages handlers for extended endpoint

kunal-mohta committed:
general endpoints for translation service and their integration tests

kunal-mohta committed:
abstract grade for nlp service; specific grade for compromise

And change google check for error with making request, to use proper error code

kunal-mohta committed:
Refactor google and yandex translation for common tasks

refactored requireData and main handler functions

kunal-mohta committed:
replace utility functions

kunal-mohta committed:
refactory mock response data - dictionary service

wrap mock responses into a single object and make changes in tests accordingly

kunal-mohta committed:
refactor mock response data - translation service

general refacor and wrapping mock responses into a single object and making changes in tests accordingly

kunal-mohta committed:
ACS-4: Dictionary Contract Tests for authentication failed cases

kunal-mohta committed:
ACS-4: Dictionary Integration tests for authentication failed cases

kunal-mohta committed:
ACS-4: Translation Integration tests for authentication failed cases

kunal-mohta committed:
Improving error handling - Dictionary service

kunal-mohta committed:
Improving error handling - NLP service

kunal-mohta committed:
Improving error handling - Translation service

kunal-mohta committed:
Extending all endpoints to have langs support

kunal-mohta committed:
ACS-3: Improving AJV log message

kunal-mohta committed:
have common character limits over services

kunal-mohta committed:
Integration tests for all endpoints extended to have langs (/langs)

kunal-mohta committed:
Update README; some tweaks to tests

kunal-mohta committed:
Integration tests (Dictionary) for Errors with making request to the service

kunal-mohta committed:
Integration Tests (Translation) for error with making request to the service

kunal-mohta committed:
Starting with middleware - implementing versioning

kunal-mohta committed:
middleware for setting response headers; NLP general refactor

kunal-mohta committed:
General refactor - dictionary

kunal-mohta committed:
General refactor - translation

kunal-mohta committed:
TravisCI config file

kunal-mohta committed:
unit test for handlerUtils; general refactor

kunal-mohta committed:
General tests refactor

kunal-mohta committed:
ACS-2: Handle new localized language codes for all services

kunal-mohta committed:
Removing the dependency of tests on environment variables

kunal-mohta committed:
Fixing oxford test grades for travis

kunal-mohta committed:
adding new npm script to run travis tasks

kunal-mohta committed:
Merge pull request #7 from kunal-mohta/travis

Travis

kunal-mohta committed:
Update checkLanguageCodes unit test to accommodate localized language codes

kunal-mohta committed:
Fixing contract tests for env variable use for tests update

kunal-mohta committed:
ACS-1: Fix for error at the end of contract tests when API keys absent

kunal-mohta committed:
General refactor and fixes in tests

kunal-mohta committed:
General Refactoring

Composition over Inheritance
